parallaxinc/PBASIC-Tokenizer
# Adds verification test runner for Emscripten PBASIC-Tokenizer compilation process

#### What's this PR do?

This adds a test runner that will execute the PBASIC test suite and validate the results of each test case.
#### Where should the reviewer start?

Setup Emscripten, run a fresh build (emscripten and webpack), execute `npm test`.
#### How should this be manually tested?

Execute the test suite, verify all tests are passing. Change values in the `EEPROM`/`PACKETBUFFER` tests to verify it will fail those output checks. Alter `DEBUG`, `TARGETMODULE`, `LANGUAGE`, `PACKETCOUNT` values in a test to verify those sanity checks will also trigger.
#### Any background context you want to provide?

`VARIABLES` check is not implemented at this time. There are currently charset issues parsing the test definition file that is causing tests #808 and #809 to fail. A new issue will be created for the charset problem.
#### What are the relevant tickets?

parallaxinc/ChromeIDE/issues/54
